src/docfx/lib/JsonUtility.cs

```diff
@@ -174,19 +174,38 @@ private static void Traverse(this JToken token, List<Error> errors, List<JToken>
                 foreach (var item in token.Children())
                 {
                     var prop = item as JProperty;
+
+                    errors = CheckFieldTypeNotExistingInSchema(type, prop, errors);
```

Comment:
>CheckFieldTypeNotExistingInSchema [](start = 29, length = 33)

this is incorrect when there are nested types, right? 

suggest to hookup `ExtensionDataSetter` delegate, sometimes we want to accept unknown keys using `JsonExtensionData` attributes.

Reply:
Tried with ExtensionDataSetter , we can filter out the mismatching field type here, but we could not get the line info for this property, according to the signature of ExtensionDataSetter, we only get property value as input not JProperty, which could cast to IJsonLineInfo

---
In reply to: [205322868](https://github.com/dotnet/docfx/pull/3112#discussion_r205322868) [](ancestors = 205322868)

Reply:
Fixed the logic by passing along with the nested type

---
In reply to: [205350969](https://github.com/dotnet/docfx/pull/3112#discussion_r205350969) [](ancestors = 205350969,205322868)